src/hooks/useMessageValidations.ts

```diff
@@ -0,0 +1,120 @@
+
+    try {
+      const fieldsToValidate: any = {
+        poly_name: "Polygon Name",
+        plantstart: "Plant Start",
+        plantend: "Plant End",
+        practice: "Restoration Practice",
+        target_sys: "Target Land Use System",
+        distr: "Tree Distribution",
+        num_trees: "Number of Trees"
+      };
+      const infoArray: ExtraInfoItem[] = JSON.parse(extraInfo);
+      return infoArray
+        .map(info => {
+          if (!info.exists) {
+            return t("{field} is missing.", { field: fieldsToValidate[info.field] });
+          } else if (info.exists && info.field === "target_sys") {
+            return t(
+              "{field}: {error} is not a valid {field} because it is not one of ['agroforest', 'natural-forest', 'mangrove', 'peatland', 'riparian-area-or-wetland', 'silvopasture', 'woodlot-or-plantation', 'urban-forest'].",
+              { field: fieldsToValidate[info.field], error: info.error }
+            );
+          } else if (info.exists && info.field === "distr") {
+            return t(
+              "{field}: {error} is not a valid {field} because it is not one of ['single-line', 'partial', 'full'].",
+              { field: fieldsToValidate[info.field], error: info.error }
+            );
+          } else if (info.exists && info.field === "num_trees") {
+            return t("{field}: {error} is not a valid {field}.", {
+              field: fieldsToValidate[info.field],
+              error: info.error
+            });
+          } else if (info.exists && info.field === "practice") {
+            return t(
+              "{field}: {error} is not a valid {field} because it is not one of ['tree-planting', 'direct-seeding', 'assisted-natural-regeneration'].",
+              { field: fieldsToValidate[info.field], error: info.error }
+            );
+          }
+          return null;
+        })
+        .filter((message): message is string => message !== null);
+    } catch {
+      return [t("Error parsing extra info.")];
+    }
+  };
+
+  const getProjectGoalMessage = (extraInfo: string | undefined) => {
+    if (!extraInfo) return [];
+    try {
+      const infoArray: ProjectGoalInfo = JSON.parse(extraInfo);
+      const { sum_area, percentage, total_area_project } = infoArray;
+      return [
+        t(
+          "Project Goal: Sum of all project polygons {sum_area} is {percentage}% of total hectares to be restored {total_area_project}",
+          { sum_area, percentage, total_area_project }
+        )
+      ];
+    } catch {
+      return [t("Error parsing extra info.")];
+    }
+  };
+  const getFormatedExtraInfo = (extraInfo: string | undefined, criteria_id: any) => {
+    if (criteria_id === 12) {
+      return getProjectGoalMessage(extraInfo);
+    } else if (criteria_id === 3) {
+      return getIntersectionMessages(extraInfo);
+    } else if (criteria_id === 14) {
+      return getDataMessage(extraInfo);
+    } else {
+      return [];
+    }
+  };
```

Comment:
since we are inside a hook, I think you should store this value with `useMemo`.
Anyways, my main concern is for you to check that this is not being called on every render, take a time to test that this is only called once per item, and not once every render.
once you send it for review again, I will test on my end to check that.

Reply:
Added the useMemo, not really shure how to test. But seems that is not calling on every render now.